src/lsp-server.ts

```diff
@@ -1123,4 +1155,49 @@ export class LspServer {
             ...workspacePreference.inlayHints ?? {}
         };
     }
+
+    async semanticTokens(params: lspsemanticTokens.SemanticTokensParams): Promise<lspsemanticTokens.SemanticTokensResult> {
```

Comment:
Can we split this into two separate methods and use types directly from the `lsp` import instead of redefining our own?